src/arrayMethodReduce.js

```diff
@@ -5,7 +5,23 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    if (arguments.length === 0 || typeof callback !== 'function') {
+      return;
```

Comment:
you return undefined if there is no callback,
according to the original method, you have an error, which is the result of calling the callback that doesn't exist;

just let this line `returnValue = callback(returnValue, this[i], i, this);` work and you will have the same errror